### PR TITLE
docs(framework:skip) Update Docker docs for 1.13.0

### PR DESCRIPTION
**Pull request by Robert-Steiner**
### Description

Updates Docker docs for 1.13.0.

### Checklist

- [ ] Implement proposed change
- [ ] Write tests
- [ ] Update [documentation](https://flower.ai/docs/writing-documentation.html)
- [ ] Make CI checks pass
- [ ] Ping maintainers on [Slack](https://flower.ai/join-slack/) (channel `#contributions`)

### CODE REVIEW

File: doc/source/docker/tutorial-quickstart-docker.rst [hunk 1]
```diff
@@ -273,47 +283,41 @@ SuperExec image and installs the required FAB dependencies.
...
+       * ``--name serverapp``: Assign the name ``serverapp`` to the container.
        * ``--detach``: Run the container in the background, freeing up the terminal.
-       * | ``flwr_superexec:0.0.1``: This is the name of the image to be run and the specific tag
+       * | ``flwr_serverapp:0.0.1``: This is the name of the image to be run and the specific tag
          | of the image.
        * | ``--insecure``: This flag tells the container to operate in an insecure mode, allowing
          | unencrypted communication.
-       * | ``--executor-config superlink=\"superlink:9091\"``: Configure the SuperExec executor to
-         | connect to the SuperLink running on port ``9091``.
+       * | ``--superlink superlink:9091``: Connect to the SuperLink's Driver API at the address
```
Comment (chongshenng) [inline doc/source/docker/tutorial-quickstart-docker.rst:319]:
```suggestion
       * | ``--superlink superlink:9091``: Connect to the SuperLink's ServerAppIO API at the address
```

File: doc/source/docker/tutorial-quickstart-docker.rst [hunk 2]
```diff
@@ -54,26 +54,36 @@ Open your terminal and run:
...
 .. dropdown:: Understand the command
 
     * ``docker run``: This tells Docker to run a container from an image.
     * ``--rm``: Remove the container once it is stopped or the command exits.
-    * | ``-p 9091:9091 -p 9092:9092``: Map port ``9091`` and ``9092`` of the container to the same port of
-      | the host machine, allowing other services to access the Driver API on
-      | ``http://localhost:9091`` and the Fleet API on ``http://localhost:9092``.
+    * | ``-p 9091:9091 -p 9092:9092 -p 9093:9093``: Map port ``9091``, ``9092`` and ``9093`` of the
+      | container to the same port of the host machine, allowing other services to access the
+      | Driver API on ``http://localhost:9091``, the Fleet API on ``http://localhost:9092`` and
```
Comment (chongshenng) [inline doc/source/docker/tutorial-quickstart-docker.rst:74]:
```suggestion
      | ServerAppIO API on ``http://localhost:9091``, the Fleet API on ``http://localhost:9092`` and
```

**Review by chongshenng — CHANGES_REQUESTED** [review]
Thanks @Robert-Steiner! Only small edits, but otherwise lgtm!

File: doc/source/docker/tutorial-quickstart-docker.rst [hunk 3]
```diff
@@ -54,26 +54,36 @@ Open your terminal and run:
...
+      | the ExecAPI on ``http://localhost:9093``.
     * ``--network flwr-network``: Make the container join the network named ``flwr-network``.
     * ``--name superlink``: Assign the name ``superlink`` to the container.
     * ``--detach``: Run the container in the background, freeing up the terminal.
     * | :substitution-code:`flwr/superlink:|stable_flwr_version|`: The name of the image to be run and the specific
       | tag of the image. The tag :substitution-code:`|stable_flwr_version|` represents a :doc:`specific version <pin-version>` of the image.
     * | ``--insecure``: This flag tells the container to operate in an insecure mode, allowing
       | unencrypted communication.
+    * | ``--executor flwr.superexec.deployment:executor`` Use the
+      | ``flwr.superexec.deployment:executor`` executor to run the ServerApp.
```
[the remark this answers is not in the document]
Comment (Robert-Steiner) [reply doc/source/docker/tutorial-quickstart-docker.rst:84]: @chongshenng  If it’s the default setting, we could consider removing it. wdyt?

[the remark this answers is not in the document]
Comment (chongshenng) [reply doc/source/docker/tutorial-quickstart-docker.rst:84]: @Robert-Steiner Yes, makes sense. Let's remove it.

[the remark this answers is not in the document]
Comment (Robert-Steiner) [reply doc/source/docker/tutorial-quickstart-docker.rst]: done

File: doc/source/docker/run-as-subprocess.rst [hunk 4]
```diff
@@ -1,53 +1,113 @@
-Run ClientApp as a Subprocess
-=============================
+Run ClientApp or ServerApp as a Subprocess
```
Comment (chongshenng) [inline doc/source/docker/run-as-subprocess.rst:1]:
```suggestion
Run ServerApp or ClientApp as a Subprocess
```

File: doc/source/docker/run-as-subprocess.rst [hunk 5]
```diff
@@ -1,53 +1,113 @@
...
+==========================================
 
-In this mode, the ClientApp is executed as a subprocess within the SuperNode Docker
-container, rather than running in a separate container. This approach reduces the number
-of running containers, which can be beneficial for environments with limited resources.
-However, it also means that the ClientApp is no longer isolated from the SuperNode,
-which may introduce additional security concerns.
+The SuperLink and SuperNode components support two distinct process modes, allowing for
```
Comment (chongshenng) [inline doc/source/docker/run-as-subprocess.rst:4]:
```suggestion
The SuperLink and SuperNode components support two distinct isolation modes, allowing for
```

File: doc/source/docker/run-as-subprocess.rst [hunk 6]
```diff
@@ -1,53 +1,113 @@
...
+flexible deployment and control:
 
-Prerequisites
--------------
+1. Subprocess Mode: In this configuration (default), the SuperLink and SuperNode take
+   responsibility for launching the Server and ClientApp processes internally. This
```
Comment (chongshenng) [inline doc/source/docker/run-as-subprocess.rst:8]:
```suggestion
   responsibility for launching the ServerApp and ClientApp processes internally. This
```

File: doc/source/docker/run-as-subprocess.rst [hunk 7]
```diff
@@ -1,53 +1,113 @@
...
+   differs from process mode ``isolation`` by using separate containers, as demonstrated
```
Comment (chongshenng) [inline doc/source/docker/run-as-subprocess.rst:9]:
```suggestion
   differs from the ``process`` isolation-mode which uses separate containers, as demonstrated
```

File: doc/source/docker/run-as-subprocess.rst [hunk 8]
```diff
@@ -1,53 +1,113 @@
...
+   in the :doc:`tutorial-quickstart-docker` guide.
 
-1. Before running the ClientApp as a subprocess, ensure that the FAB dependencies have
-   been installed in the SuperNode images. This can be done by extending the SuperNode
-   image:
+   This approach reduces the number of running containers, which can be beneficial for
```
Comment (chongshenng) [inline doc/source/docker/run-as-subprocess.rst:12]:
```suggestion
   Using the ``subprocess`` approach reduces the number of running containers, which can be beneficial for
```

File: doc/source/docker/run-as-subprocess.rst [hunk 9]
```diff
@@ -1,53 +1,113 @@
...
+   environments with limited resources. However, it also means that the applications are
+   not isolated from their parent containers, which may introduce additional security
+   concerns.
 
-   .. code-block:: dockerfile
-       :caption: Dockerfile.supernode
-       :linenos:
-       :substitutions:
+2. Isolation Mode: In this mode, the Server and ClientApp run in completely separate
```
Comment (chongshenng) [inline doc/source/docker/run-as-subprocess.rst:17]:
```suggestion
2. Process Mode: In this mode, the ServerApp and ClientApp run in completely separate
```

File: doc/source/docker/run-as-subprocess.rst [hunk 10]
```diff
@@ -1,53 +1,113 @@
...
+   processes. Unlike the alternative Subprocess mode, the SuperLink or SuperNode does
+   not attempt to create or manage these processes. Instead, they must be started
+   externally.
 
-       FROM flwr/supernode:|stable_flwr_version|
+These modes can be mixed for added flexibility. For instance, you can run the SuperLink
```
Comment (chongshenng) [inline doc/source/docker/run-as-subprocess.rst:22]:
```suggestion
Both modes can be mixed for added flexibility. For instance, you can run the SuperLink
```

File: doc/source/docker/run-as-subprocess.rst [hunk 11]
```diff
@@ -1,53 +1,113 @@
...
+in ``subprocess`` mode while keeping the SuperNode in ``isolation`` mode, or vice versa.
```
Comment (chongshenng) [inline doc/source/docker/run-as-subprocess.rst:23]:
```suggestion
in ``subprocess`` mode while keeping the SuperNode in ``process`` mode, or vice versa.
```

File: doc/source/docker/run-as-subprocess.rst [hunk 12]
```diff
@@ -1,53 +1,113 @@
...
+It is also possible to configure both to run entirely in ``isolation`` mode, depending
+on your specific requirements.
```
Comment (chongshenng) [inline doc/source/docker/run-as-subprocess.rst:25]: I think this sentence is now unnecessary with the changes above.

File: doc/source/docker/run-as-subprocess.rst [hunk 13]
```diff
@@ -1,53 +1,113 @@
...
 
-       WORKDIR /app
-       COPY pyproject.toml .
-       RUN sed -i 's/.*flwr\[simulation\].*//' pyproject.toml \
-           && python -m pip install -U --no-cache-dir .
+To run the SuperLink and SuperNode in process mode ``isolation``, refer to the
```
Comment (chongshenng) [inline doc/source/docker/run-as-subprocess.rst:27]:
```suggestion
To run the SuperLink and SuperNode in isolation mode ``process``, refer to the
```

File: doc/source/docker/run-as-subprocess.rst [hunk 14]
```diff
@@ -1,53 +1,113 @@
...
+        2. Next, build the SuperNode Docker image by running the following command in the
+        directory where Dockerfile is located:
+
+        .. code-block:: shell
+
+            $ docker build -f supernode.Dockerfile -t flwr_supernode:0.0.1 .
+
+        **Run the ClientApp as a Subprocess**
+
+        Start the SuperNode and run the ClientApp as a subprocess:
```
Comment (chongshenng) [inline doc/source/docker/run-as-subprocess.rst:64]:
```suggestion
        Start the SuperNode and run the ClientApp as a subprocess (note that the subprocess mode is the default, so you do not have to explicitly set the ``--isolation`` flag:
```

File: doc/source/docker/tutorial-quickstart-docker.rst [hunk 15]
```diff
@@ -54,29 +54,36 @@ Open your terminal and run:
...
 
     * ``docker run``: This tells Docker to run a container from an image.
     * ``--rm``: Remove the container once it is stopped or the command exits.
-    * | ``-p 9091:9091 -p 9092:9092``: Map port ``9091`` and ``9092`` of the container to the same port of
-      | the host machine, allowing other services to access the Driver API on
-      | ``http://localhost:9091`` and the Fleet API on ``http://localhost:9092``.
+    * | ``-p 9091:9091 -p 9092:9092 -p 9093:9093``: Map port ``9091``, ``9092`` and ``9093`` of the
+      | container to the same port of the host machine, allowing other services to access the
+      | ServerAppIO API on ``http://localhost:9091``, the Fleet API on ``http://localhost:9092`` and
+      | the ExecAPI on ``http://localhost:9093``.
```
Comment (chongshenng) [inline doc/source/docker/tutorial-quickstart-docker.rst:73]:
```suggestion
      | the Exec API on ``http://localhost:9093``.
```

File: doc/source/docker/tutorial-quickstart-docker.rst [hunk 16]
```diff
@@ -107,16 +114,16 @@ Start two SuperNode containers.
...
        * | ``--insecure``: This flag tells the container to operate in an insecure mode, allowing
          | unencrypted communication.
        * | ``--superlink superlink:9092``: Connect to the SuperLink's Fleet API at the address
          | ``superlink:9092``.
        * | ``--node-config "partition-id=0 num-partitions=2"``: Set the partition ID to ``0`` and the
          | number of partitions to ``2`` for the SuperNode configuration.
-       * | ``--supernode-address 0.0.0.0:9094``: Set the address and port number that the SuperNode
-         | is listening on.
+       * | ``--clientappio-api-address 0.0.0.0:9094``: Set the address and port number that the
+         | SuperNode is listening on to communicate with the ClientApp.
```
Comment (chongshenng) [inline doc/source/docker/tutorial-quickstart-docker.rst:126]:
```suggestion
         | SuperNode is listening on to communicate with the ClientApp. If 
         | two SuperNodes are started on the same machine, set two different port numbers for each SuperNode.
         | (E.g. In the next step, we set the second SuperNode container to listen on port 9095)
```

File: doc/source/docker/enable-tls.rst [hunk 17]
```diff
@@ -14,139 +11,323 @@ and a PEM-encoded certificate chain.
...
+
+    For example, to change the user ID of all files in the ``certificates/`` directory,
+    you can run ``sudo chown -R 49999:49999 certificates/*``.
+
+    If you later want to delete the directory, you can change the user ID back to the
+    current user ID by running ``sudo chown -R $USER:$(id -gn) certificates``.
+
+.. tab-set::
+
+    .. tab-item:: Enable TLS in Process Mode ``subprocess``
```
Comment (chongshenng) [inline doc/source/docker/enable-tls.rst:27]:
```suggestion
    .. tab-item:: Enable TLS in Isolation Mode ``subprocess``
```

File: doc/source/docker/enable-tls.rst [hunk 18]
```diff
@@ -14,139 +11,323 @@ and a PEM-encoded certificate chain.
...
+
+    .. tab-item:: Enable TLS in Process Mode ``isolation``
+
+        In process mode ``isolation``, the Server and ClientApp run in their own processes.
+        Unlike in process mode ``subprocess``, the SuperLink or SuperNode does not attempt to
+        create the respective processes; instead, they must be created externally.
+
+        It is possible to run only the SuperLink in process mode ``subprocess`` and the
+        SuperNode in process mode ``isolation``, or vice versa, or even both with process mode
+        ``isolation``.
```
Comment (chongshenng) [inline doc/source/docker/enable-tls.rst:147]:
```suggestion
        It is possible to run only the SuperLink in isolation mode ``subprocess`` and the
        SuperNode in isolation mode ``process``, or vice versa, or even both with isolation mode
        ``process``.
```

File: doc/source/docker/run-as-subprocess.rst (hunk 12 above)
Comment (Robert-Steiner) [reply doc/source/docker/run-as-subprocess.rst]: removed

**Comment by Robert-Steiner** [comment]
@chongshenng, thanks for the careful review. I somehow mixed up the `isolation` flag with the `process` mode. Sorry about that!

File: doc/source/docker/enable-tls.rst [hunk 19]
```diff
@@ -14,139 +11,323 @@ and a PEM-encoded certificate chain.
...
+
+            $ docker run --rm \
+                --volume ./supernode-certificates/:/app/certificates/:ro \
+                --volume ./superlink-certificates/ca.crt:/app/ca.crt/:ro \
+                <supernode-image> \
+                --ssl-ca-certfile certificates/ca.crt \
+                --ssl-certfile certificates/server.pem \
+                --ssl-keyfile certificates/server.key \
+                --root-certificates ca.crt \
+                <additional-args>
```
Comment (chongshenng) [inline doc/source/docker/enable-tls.rst:103]:
```suggestion
            $ docker run --rm \
                --volume ./superlink-certificates/ca.crt:/app/ca.crt/:ro \
                <supernode-image> \
                --root-certificates ca.crt \
                <additional-args>
```

File: doc/source/docker/enable-tls.rst [hunk 20]
```diff
@@ -14,139 +11,323 @@ and a PEM-encoded certificate chain.
...
+        .. dropdown:: Understanding the command
+
+            * ``docker run``: This tells Docker to run a container from an image.
+            * ``--rm``: Remove the container once it is stopped or the command exits.
+            * | ``--volume ./supernode-certificates/:/app/certificates/:ro``: Mount the ``supernode-certificates``
+              | directory in the current working directory of the host machine as a read-only volume at the
+              | ``/app/certificates`` directory inside the container.
+              |
+              | This allows the container to access the TLS certificates that are stored in the certificates
+              | directory.
```
Comment (chongshenng) [inline doc/source/docker/enable-tls.rst:114]:
```suggestion
```

File: doc/source/docker/enable-tls.rst [hunk 21]
```diff
@@ -14,139 +11,323 @@ and a PEM-encoded certificate chain.
...
+            * | ``--ssl-certfile certificates/server.pem``: Specify the location of the SuperNode's
+              | TLS certificate file inside the container.
+              |
+              | The ``certificates/server.pem`` file is used to identify the SuperNode and to encrypt the
+              | data that is transmitted over the network.
+            * | ``--ssl-keyfile certificates/server.key``: Specify the location of the SuperNode's
+              | TLS private key file inside the container.
+              |
+              | The ``certificates/server.key`` file is used to decrypt the data that is transmitted over
+              | the network.
```
Comment (chongshenng) [inline doc/source/docker/enable-tls.rst:133]:
```suggestion
```

File: doc/source/docker/enable-tls.rst [hunk 22]
```diff
@@ -14,139 +11,323 @@ and a PEM-encoded certificate chain.
...
+        ``process``.
+
+        **SuperLink and ServerApp**
+
+        To enable TLS between all Flower components, you will need two sets of PEM-encoded root
+        certificates, private keys, and certificate chains.
+
+        Assuming all files we need are in the local ``superlink-certificates`` and
+        ``supernode-certificates`` directories, we can use the flag ``--volume`` to mount the
+        local directories into the SuperNode container:
```
Comment (chongshenng) [inline doc/source/docker/enable-tls.rst:156]:
```suggestion
        Assuming all files we need are in the local ``superlink-certificates``
        directory, we can use the flag ``--volume`` to mount the
        local directory into the SuperLink container:
```

File: doc/source/docker/enable-tls.rst [hunk 23]
```diff
@@ -14,139 +11,323 @@ and a PEM-encoded certificate chain.
...
+
+        Start the SuperLink container:
```
Comment (chongshenng) [inline doc/source/docker/enable-tls.rst:158]:
```suggestion
```

We probably don't need this line anymore.

File: doc/source/docker/enable-tls.rst [hunk 24]
```diff
@@ -14,139 +11,323 @@ and a PEM-encoded certificate chain.
...
+              | independent process. The SuperLink does not attempt to create it.
+
+        Start the ServerApp container:
+
+        .. code-block:: bash
+
+            $ docker run --rm \
+                --volume ./superlink-certificates/ca.crt:/app/ca.crt:ro \
+                <serverapp-image> \
+                --root-certificates ca.crt \
```
Comment (chongshenng) [inline doc/source/docker/enable-tls.rst:209]:
```suggestion
                <serverapp-image> \
                --insecure \
```

File: doc/source/docker/enable-tls.rst [hunk 25]
```diff
@@ -14,139 +11,323 @@ and a PEM-encoded certificate chain.
...
+                <additional-args>
+
+        .. dropdown:: Understand the command
+
+            * ``docker run``: This tells Docker to run a container from an image.
+            * ``--rm``: Remove the container once it is stopped or the command exits.
+            * | ``--volume ./superlink-certificates/ca.crt:/app/ca.crt:ro``: Mount the ``ca.crt`` file from
+              | the ``superlink-certificates`` directory of the host machine as a read-only volume at the
+              | ``/app/ca.crt`` directory inside the container.
```
Comment (chongshenng) [inline doc/source/docker/enable-tls.rst:218]:
```suggestion
```

File: doc/source/docker/enable-tls.rst [hunk 26]
```diff
@@ -14,139 +11,323 @@ and a PEM-encoded certificate chain.
...
+            * ``<serverapp-image>``: The name of your ServerApp image to be run.
+            * | ``--root-certificates ca.crt``: This specifies the location of the CA
+              | certificate file inside the container.
+              |
+              | The ``ca.crt`` file is used to verify the identity of the SuperLink.
```
Comment (chongshenng) [inline doc/source/docker/enable-tls.rst:223]:
```suggestion
            * | ``--insecure``:  This flag tells the container to operate in an insecure mode, allowing
         | unencrypted communication. Secure connections will be added in future releases.
```

File: doc/source/docker/enable-tls.rst [hunk 27]
```diff
@@ -14,139 +11,323 @@ and a PEM-encoded certificate chain.
...
+        .. code-block:: bash
+            :substitutions:
+
+            $ docker run --rm \
+                --volume ./supernode-certificates/:/app/certificates/:ro \
+                --volume ./superlink-certificates/ca.crt:/app/ca.crt/:ro \
+                flwr/supernode:|stable_flwr_version| \
+                --ssl-ca-certfile=certificates/ca.crt \
+                --ssl-certfile=certificates/server.pem \
+                --ssl-keyfile=certificates/server.key \
```
Comment (chongshenng) [inline doc/source/docker/enable-tls.rst:249]:
```suggestion
```

File: doc/source/docker/enable-tls.rst [hunk 28]
```diff
@@ -14,139 +11,323 @@ and a PEM-encoded certificate chain.
...
+            Each SuperNode can have its own set of keys and certificates, or they can all share
+            the same set.
+
+        Start the SuperNode container:
+
+        .. code-block:: bash
+            :substitutions:
+
+            $ docker run --rm \
+                --volume ./supernode-certificates/:/app/certificates/:ro \
```
Comment (chongshenng) [inline doc/source/docker/enable-tls.rst:244]:
```suggestion
```

File: doc/source/docker/enable-tls.rst [hunk 29]
```diff
@@ -14,139 +11,323 @@ and a PEM-encoded certificate chain.
...
+              | certificate file inside the container.
+              |
+              | The ``ca.crt`` file is used to verify the identity of the SuperLink.
+
+        **SuperNode and ClientApp**
+
+        .. note::
+
+            If you're generating self-signed certificates and the ``ca.crt`` certificate or the
+            ``supernode-certificates`` directory doesn't exist on the SuperNode, you can copy it over
```
Comment (chongshenng) [inline doc/source/docker/enable-tls.rst:230]:
```suggestion
            If you're generating self-signed certificates and the ``ca.crt`` certificate 
            doesn't exist on the SuperNode, you can copy it over
```

File: doc/source/docker/enable-tls.rst [hunk 30]
```diff
@@ -14,139 +11,323 @@ and a PEM-encoded certificate chain.
...
+            after the generation step.
+
+        .. note::
+
+            Each SuperNode can have its own set of keys and certificates, or they can all share
+            the same set.
```
Comment (chongshenng) [inline doc/source/docker/enable-tls.rst:236]:
```suggestion
            Each SuperNode can have its own set of certificates, or they can all share
            the same set.
```

File: doc/source/docker/enable-tls.rst [hunk 31]
```diff
@@ -14,139 +11,323 @@ and a PEM-encoded certificate chain.
...
+        .. dropdown:: Understanding the command
+
+            * ``docker run``: This tells Docker to run a container from an image.
+            * ``--rm``: Remove the container once it is stopped or the command exits.
+            * | ``--volume ./supernode-certificates/:/app/certificates/:ro``: Mount the ``supernode-certificates``
+              | directory in the current working directory of the host machine as a read-only volume at the
+              | ``/app/certificates`` directory inside the container.
+              |
+              | This allows the container to access the TLS certificates that are stored in the certificates
+              | directory.
```
Comment (chongshenng) [inline doc/source/docker/enable-tls.rst:263]:
```suggestion
```

File: doc/source/docker/enable-tls.rst [hunk 32]
```diff
@@ -14,139 +11,323 @@ and a PEM-encoded certificate chain.
...
+            * | ``--ssl-certfile certificates/server.pem``: Specify the location of the SuperNode's
+              | TLS certificate file inside the container.
+              |
+              | The ``certificates/server.pem`` file is used to identify the SuperNode and to encrypt the
+              | data that is transmitted over the network.
+            * | ``--ssl-keyfile certificates/server.key``: Specify the location of the SuperNode's
+              | TLS private key file inside the container.
+              |
+              | The ``certificates/server.key`` file is used to decrypt the data that is transmitted over
+              | the network.
```
Comment (chongshenng) [inline doc/source/docker/enable-tls.rst:283]:
```suggestion
```

File: doc/source/docker/enable-tls.rst [hunk 33]
```diff
@@ -14,139 +11,323 @@ and a PEM-encoded certificate chain.
...
+              | independent process. The SuperNode does not attempt to create it.
+
+        Start the ClientApp container:
+
+        .. code-block:: bash
+
+            $ docker run --rm \
+                --volume ./supernode-certificates/ca.crt:/app/ca.crt:ro \
+                <clientapp-image> \
+                --root-certificates ca.crt \
```
Comment (chongshenng) [inline doc/source/docker/enable-tls.rst:298]:
```suggestion
                <clientapp-image> \
                --insecure \
```

File: doc/source/docker/enable-tls.rst [hunk 34]
```diff
@@ -14,139 +11,323 @@ and a PEM-encoded certificate chain.
...
+                <additional-args>
+
+        .. dropdown:: Understand the command
+
+            * ``docker run``: This tells Docker to run a container from an image.
+            * ``--rm``: Remove the container once it is stopped or the command exits.
+            * | ``--volume ./supernode-certificates/ca.crt:/app/ca.crt:ro``: Mount the ``ca.crt`` file from
+              | the ``supernode-certificates`` directory of the host machine as a read-only volume at the
+              | ``/app/ca.crt`` directory inside the container.
```
Comment (chongshenng) [inline doc/source/docker/enable-tls.rst:307]:
```suggestion
```

File: doc/source/docker/enable-tls.rst [hunk 35]
```diff
@@ -14,139 +11,323 @@ and a PEM-encoded certificate chain.
...
+            * ``<clientapp-image>``: The name of your ClientApp image to be run.
+            * | ``--root-certificates ca.crt``: This specifies the location of the CA
+              | certificate file inside the container.
+              |
+              | The ``ca.crt`` file is used to verify the identity of the SuperNode.
```
Comment (chongshenng) [inline doc/source/docker/enable-tls.rst:312]:
```suggestion
            * | ``--insecure``:  This flag tells the container to operate in an insecure mode, allowing
         | unencrypted communication. Secure connections will be added in future releases.
```

File: doc/source/docker/run-as-subprocess.rst [hunk 36]
```diff
@@ -1,53 +1,113 @@
...
+
+        .. code-block:: shell
+
+            $ docker run --rm \
+                --detach \
+                flwr_supernode:0.0.1 \
+                --insecure \
+                --superlink <superlink-address>:9092
+
+    .. tab-item:: ServerApp
```
Comment (chongshenng) [inline doc/source/docker/run-as-subprocess.rst:73]: Let's swap the order and have `ServerApp` at the top, followed by `ClientApp`. This makes it consistent with the title.

File: doc/source/docker/tutorial-quickstart-docker.rst [hunk 37]
```diff
@@ -209,18 +218,21 @@ required dependencies.
...
+           --clientappio-api-address supernode-1:9094
 
    .. dropdown:: Understand the command
 
        * ``docker run``: This tells Docker to run a container from an image.
        * ``--rm``: Remove the container once it is stopped or the command exits.
        * ``--network flwr-network``: Make the container join the network named ``flwr-network``.
        * ``--detach``: Run the container in the background, freeing up the terminal.
+       * | ``--insecure``: This flag tells the container to operate in an insecure mode, allowing
+         | unencrypted communication.
```
Comment (chongshenng) [inline doc/source/docker/tutorial-quickstart-docker.rst:231]:
```suggestion
         | unencrypted communication. Secure connections will be added in future releases.
```

File: doc/source/docker/tutorial-quickstart-docker.rst [hunk 38]
```diff
@@ -273,47 +286,41 @@ SuperExec image and installs the required FAB dependencies.
...
-         | the host machine, allowing you to access the SuperExec API on ``http://localhost:9093``.
        * ``--network flwr-network``: Make the container join the network named ``flwr-network``.
-       * ``--name superexec``: Assign the name ``superexec`` to the container.
+       * ``--name serverapp``: Assign the name ``serverapp`` to the container.
        * ``--detach``: Run the container in the background, freeing up the terminal.
-       * | ``flwr_superexec:0.0.1``: This is the name of the image to be run and the specific tag
+       * | ``flwr_serverapp:0.0.1``: This is the name of the image to be run and the specific tag
          | of the image.
        * | ``--insecure``: This flag tells the container to operate in an insecure mode, allowing
          | unencrypted communication.
```
Comment (chongshenng) [inline doc/source/docker/tutorial-quickstart-docker.rst:321]:
```suggestion
         | unencrypted communication. Secure connections will be added in future releases.
```

File: doc/source/docker/tutorial-quickstart-docker.rst [hunk 39]
```diff
@@ -230,38 +242,39 @@ required dependencies.
            --network flwr-network \
            --detach \
            flwr_clientapp:0.0.1 \
-           --supernode supernode-2:9095
+           --insecure \
+           --clientappio-api-address supernode-2:9095
 
-Step 5: Start the SuperExec
----------------------------
+Step 5: Start a ServerApp
```
Comment (chongshenng) [inline doc/source/docker/tutorial-quickstart-docker.rst:248]: Can we have "Step 4: Start a ServerApp" and "Step 5: Start a ClientApp"?

File: doc/source/docker/tutorial-quickstart-docker.rst [hunk 40]
```diff
@@ -347,17 +354,25 @@ Step 7: Update the Application
        partition_train_test = partition.train_test_split(test_size=0.2, seed=43)
        # ...
 
-2. Stop the current ClientApp containers:
+2. Stop the current ClientApp and ServerApp containers:
```
Comment (chongshenng) [inline doc/source/docker/tutorial-quickstart-docker.rst:357]:
```suggestion
2. Stop the current ServerApp and ClientApp containers:
```

File: doc/source/docker/tutorial-quickstart-docker.rst [hunk 41]
```diff
@@ -347,17 +354,25 @@ Step 7: Update the Application
...
 
    .. code-block:: bash
 
-       $ docker stop $(docker ps -a -q  --filter ancestor=flwr_clientapp:0.0.1)
+       $ docker stop $(docker ps -a -q  --filter ancestor=flwr_clientapp:0.0.1) serverapp
+
+3. Rebuild ClientApp and ServerApp image:
```
Comment (chongshenng) [inline doc/source/docker/tutorial-quickstart-docker.rst:363]:
```suggestion
3. Rebuild ServerApp and ClientApp images:
```

File: doc/source/docker/tutorial-quickstart-docker.rst [hunk 42]
```diff
@@ -347,17 +354,25 @@ Step 7: Update the Application
...
+       If you have modified the dependencies listed in your ``pyproject.toml`` file, it
+       is essential to rebuild images.
+
+       If you haven't made any changes, you can skip this step.
 
    .. code-block:: bash
 
-       $ docker build -f Dockerfile.clientapp -t flwr_clientapp:0.0.1 .
+       $ docker build -f clientapp.Dockerfile -t flwr_clientapp:0.0.1 . && \
+         docker build -f serverapp.Dockerfile -t flwr_serverapp:0.0.1 .
```
Comment (chongshenng) [inline doc/source/docker/tutorial-quickstart-docker.rst:375]:
```suggestion
       $ docker build -f clientapp.Dockerfile -t flwr_serverapp:0.0.1 . && \
         docker build -f serverapp.Dockerfile -t flwr_clientapp:0.0.1 .
```

File: doc/source/docker/tutorial-quickstart-docker.rst [hunk 43]
```diff
@@ -367,12 +382,21 @@ Step 7: Update the Application
...
-           --supernode supernode-2:9095
+           --insecure \
+           --clientappio-api-address supernode-2:9095
+       $ docker run --rm \
+           --network flwr-network \
+           --name serverapp \
+           --detach \
+           flwr_serverapp:0.0.1 \
+           --insecure \
+           ----serverappio-api-address superlink:9091
```
Comment (chongshenng) [inline doc/source/docker/tutorial-quickstart-docker.rst:399]: Can you again start with the ServerApp, followed by the two ClientApps? This makes it consistent: we always start with the SuperLink (ServerApp) then the SuperNodes(ClientApps).

**Review by chongshenng — CHANGES_REQUESTED** [review]
Thanks for your patience @Robert-Steiner. Main comments come from the recent changes to `flwr-serverapp` and `flwr-clientapp` CLI arguments.

**Review by chongshenng — APPROVED** [review]
LGTM @Robert-Steiner !

**Review by danieljanes — APPROVED** [review]

**Pull request merged**